app/tests/integration/nimbus/test_desktop_client_integrations.py

```diff
@@ -190,3 +199,107 @@ def test_check_telemetry_enrollment_unenrollment(
     else:
         if control is not False:
             assert False, "Experiment unenrollment was never seen in Ping Data"
+
+
+@pytest.mark.nimbus_integration
+@pytest.mark.xdist_group(name="group2")
+def test_check_telemetry_pref_flip(
+    base_url,
+    selenium,
+    kinto_client,
+    slugify,
+    experiment_name,
+    create_desktop_experiment,
+    experiment_default_data,
+    check_ping_for_experiment,
+    telemetry_event_check,
+):
+    _row_locator = (By.CSS_SELECTOR, "tr > td > span > span")
+    _search_bar_locator = (By.ID, "about-config-search")
+
+    requests.delete("http://ping-server:5000/pings")
+    targeting = helpers.load_targeting_configs()[0]
+    experiment_slug = str(slugify(experiment_name))
+    experiment_default_data["targetingConfigSlug"] = targeting
+    experiment_default_data["featureConfigId"] = 9
+    experiment_default_data["referenceBranch"] = {
+        "description": "reference branch",
+        "name": "Branch 1",
+        "ratio": 99,
+        "featureEnabled": True,
+        "featureValue": '{"value": "test_string_automation"}',
+    }
+    experiment_default_data["treatmentBranches"] = [
+        {
+            "description": "treatment branch",
+            "name": "Branch 2",
+            "ratio": 1,
+            "featureEnabled": False,
+            "featureValue": "",
+        }
+    ]
```

Comment:
Oh I see here, you're trying to force it into the reference branch, you can just empty out the treatment branches altogether if you don't want to use it.  This might introduce intermittency if it happens to enroll in this other branch, we should probably take this out.

Reply:
So since I am using graphQL to create the experiment, if I don't give it a treatment branch it won't create the experiment.

Reply:
What do you mean? What's the error?  Seems to work locally:

```py
        response = self.query(
            UPDATE_EXPERIMENT_MUTATION,
            variables={
                "input": {
                    "id": experiment.id,
                    "hypothesis": "Test Hypothesis",
                    "application": "DESKTOP",
                    "changelogMessage": "test updates",
                    "publicDescription": "Some sort of Fancy Words",
                    "riskRevenue": False,
                    "riskPartnerRelated": False,
                    "riskBrand": False,
                    "featureConfigId": 1,
                    "referenceBranch": {
                        "description": "reference branch",
                        "name": "Branch 1",
                        "ratio": 50,
                        "featureEnabled": True,
                        "featureValue": "{}",
                    },
                    "treatmentBranches": [],
                    "populationPercent": "100",
                    "totalEnrolledClients": 55,
                }
            },
            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
        )


ipdb> response.content
b'{"data":{"updateExperiment":{"message":"success"}}}'
ipdb> experiment = NimbusExperiment.objects.get()
ipdb> experiment.reference_branch
<NimbusBranch: old name: Branch 1>
ipdb> experiment.treatment_branches
[]
```

Reply:
I take that back, it works fine